datafusion/sqllogictest/test_files/union.slt

```diff
@@ -836,3 +836,18 @@ physical_plan
 # Clean up after the test
 statement ok
 drop table aggregate_test_100;
+
+# test for https://github.com/apache/datafusion/issues/14352
```

Comment:
I verified this test fails without the code in this PR:

```diff
External error: query result mismatch:
[SQL] SELECT
    a,
    a IS NOT NULL
FROM (
    -- second column, even though it's not selected, was necessary to reproduce the bug linked above
    SELECT 'foo' AS a, 3 AS b
    UNION ALL
    SELECT NULL AS a, 4 AS b
)
[Diff] (-expected|+actual)
-   NULL false
+   NULL true
    foo true
at test_files/union.slt:841
```

Reply:
thanks for checking that!